drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,359 @@
+		return err
+	}
+
+	cp.Namespace = storageNs
+	cp.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	if cp.Spec.Backend.Service != nil {
+		cp.Spec.Backend.Service.Namespace = storageNs
+	}
+
+	if len(cp.Spec.Proxy) > 0 && cp.Spec.Proxy[0].Endpoint.Service != nil {
+		cp.Spec.Proxy[0].Endpoint.Service.Namespace = storageNs
+	}
+
+	existingPlugin := &console.ConsolePlugin{}
+	err := p.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      cp.Name,
+			Namespace: cp.Namespace,
+		},
+		existingPlugin,
+	)
+
+	if errors.IsNotFound(err) {
+		logrus.Infof("Creating %s Consoleplugin", cp.Name)
+		return p.client.Create(context.TODO(), cp)
+	} else if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *plugin) deleteConsolePlugin(name, storageNs string, owners ...metav1.OwnerReference) error {
+	if err := console.AddToScheme(p.scheme); err != nil {
+		return err
+	}
+
+	resource := types.NamespacedName{
+		Name:      name,
+		Namespace: storageNs,
+	}
+
+	consolePlugin := &console.ConsolePlugin{}
+	err := p.client.Get(context.TODO(), resource, consolePlugin)
+	if errors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	newOwners := k8s.RemoveOwners(consolePlugin.OwnerReferences, owners)
+
+	// Do not delete the object if it does not have the owner that was passed;
+	// even if the object has no owner
+	if (len(consolePlugin.OwnerReferences) == 0 && len(owners) > 0) ||
+		(len(consolePlugin.OwnerReferences) > 0 && len(consolePlugin.OwnerReferences) == len(newOwners)) {
+		logrus.Infof("Cannot delete consolePlugin %s/%s as it is not owned",
+			storageNs, name)
+		return nil
+	}
+
+	if len(newOwners) == 0 {
+		logrus.Infof("Deleting %s/%s consolePlugin", storageNs, name)
+		return p.client.Delete(context.TODO(), consolePlugin)
+	}
+	consolePlugin.OwnerReferences = newOwners
+	logrus.Infof("Disowning %s/%s consolePlugin", storageNs, name)
+	return p.client.Update(context.TODO(), consolePlugin)
+}
+
+func updateDataIfNginxConfigMap(cm *v1.ConfigMap, storageNs string) {
+	if cm.Name == NginxConfigMapName {
+		cm.Data = map[string]string{
+			"nginx.conf": `pid /tmp/nginx.pid;
+    events {
+      worker_connections 1024;
+    }
+    http {
+      server {
+        listen 8080;
+          server_name portworx-console-proxy.` + storageNs + `.svc.cluster.local;
+        location / {
+          proxy_pass http://portworx-api.` + storageNs + `.svc.cluster.local:9021;
+        }
+      }
+      server {
+        listen 8443 ssl;
+        server_name portworx-console-proxy.` + storageNs + `.svc.cluster.local;
+        ssl_certificate /etc/nginx/certs/tls.crt;
+        ssl_certificate_key /etc/nginx/certs/tls.key;
+        location / {
+          proxy_pass http://portworx-api.` + storageNs + `.svc.cluster.local:9021;
+        }
+      }
+    }`,
+		}
+	}
+}
+
+func isVersionGreaterOrEqual(version string, targetVersion string) bool {
```

Comment:
Use `version "github.com/hashicorp/go-version"` package to parse and compare versions. Search for the usage in the operator code and you will find examples.